criu/mount.c

```diff
@@ -765,6 +780,9 @@ static int resolve_external_mounts(struct mount_info *info)
 		if (m->parent == NULL || m->is_ns_root)
 			continue;
 
+		/* Check mount has external sharing */
+		m->internal_sharing = !has_external_sharing(ext_ns->mnt.mntinfo_list, m);
```

Comment:
what will has_external_sharing return on restore? This idea to look up external sharing in a criu namespace looks wrong for me. On dump, we can look up a mount. But how it should work on restore is unclear for me. What if we will restore on another host where shared, master ids for mounts are completely different?

Reply:
>what will has_external_sharing return on restore?
These function is called on dump only.
>This idea to look up external sharing in a criu namespace looks wrong for me. On dump, we can look up a mount. But how it should work on restore is unclear for me. What if we will restore on another host where shared, master ids for mounts are completely different?
On restore one should use m->internal_sharing same as it was before.

Sorry if I miss something.

Reply:
> These function is called on dump only.
Oops. I misread the for_dump argument. I still don't understand the idea of resolve_external_mounts. Let's think that we don't use any auto ext mount options. We explicitly specify external mounts. In this case, I don't understand why we need to enumerate all mounts in the criu namespace.

Reply:
You miss that I'm trying to detect external _sharing_ not external mounts.

1) Even if you know that the mount is manually marked as external mount if it is shared it has two options: a)external share (need to inherit share) and b) internal share (need to create own share).

2) If "user" does not mark the mount as external mount it still can (or not) have external sharing, we need to detect this external dependecy and e.g. fail as we can't resolve external deps without helpof the "user".